lib/url.js

```diff
@@ -948,13 +953,23 @@ Url.prototype.resolveObject = function resolveObject(relative) {
 
 Url.prototype.parseHost = function parseHost() {
   var host = this.host;
-  var port = portPattern.exec(host);
-  if (port) {
-    port = port[0];
-    if (port !== ':') {
-      this.port = port.slice(1);
+  var port;
+  // Adding support for the unix socket hostname and port resolution
```

Comment:
How about ref #30002 here to help others to read?

Reply:
```suggestion
  // Unix socket hostname and port resolution
```